src/data/heal_locations_pkm_center.h

```diff
@@ -0,0 +1,171 @@
+        .map = MAP_NUM(SOOTOPOLIS_CITY_POKEMON_CENTER_1F),
+        DEFAULT_POKEMON_CENTER_COORDS,
+    },
+    [HEAL_LOCATION_EVER_GRANDE_CITY - 1] =
+    {
+        .group = MAP_GROUP(EVER_GRANDE_CITY_POKEMON_CENTER_1F),
+        .map = MAP_NUM(EVER_GRANDE_CITY_POKEMON_CENTER_1F),
+        DEFAULT_POKEMON_CENTER_COORDS,
+    },
+    [HEAL_LOCATION_LITTLEROOT_TOWN_BRENDANS_HOUSE - 1] =
+    {
+        .group = MAP_GROUP(LITTLEROOT_TOWN_BRENDANS_HOUSE_1F),
+        .map = MAP_NUM(LITTLEROOT_TOWN_BRENDANS_HOUSE_1F),
+        .x = 2,
+        .y = 7,
+    },
+    [HEAL_LOCATION_LITTLEROOT_TOWN_MAYS_HOUSE - 1] =
+    {
+        .group = MAP_GROUP(LITTLEROOT_TOWN_MAYS_HOUSE_1F),
+        .map = MAP_NUM(LITTLEROOT_TOWN_MAYS_HOUSE_1F),
+        .x = 8,
+        .y = 7,
+    },
+    [HEAL_LOCATION_OLDALE_TOWN - 1] =
+    {
+        .group = MAP_GROUP(OLDALE_TOWN_POKEMON_CENTER_1F),
+        .map = MAP_NUM(OLDALE_TOWN_POKEMON_CENTER_1F),
+        DEFAULT_POKEMON_CENTER_COORDS,
+    },
+    [HEAL_LOCATION_DEWFORD_TOWN - 1] =
+    {
+        .group = MAP_GROUP(DEWFORD_TOWN_POKEMON_CENTER_1F),
+        .map = MAP_NUM(DEWFORD_TOWN_POKEMON_CENTER_1F),
+        DEFAULT_POKEMON_CENTER_COORDS,
+    },
+    [HEAL_LOCATION_LAVARIDGE_TOWN - 1] =
+    {
+        .group = MAP_GROUP(LAVARIDGE_TOWN_POKEMON_CENTER_1F),
+        .map = MAP_NUM(LAVARIDGE_TOWN_POKEMON_CENTER_1F),
+        DEFAULT_POKEMON_CENTER_COORDS,
+    },
+    [HEAL_LOCATION_FALLARBOR_TOWN - 1] =
+    {
+        .group = MAP_GROUP(FALLARBOR_TOWN_POKEMON_CENTER_1F),
+        .map = MAP_NUM(FALLARBOR_TOWN_POKEMON_CENTER_1F),
+        DEFAULT_POKEMON_CENTER_COORDS,
+    },
+    [HEAL_LOCATION_VERDANTURF_TOWN - 1] =
+    {
+        .group = MAP_GROUP(VERDANTURF_TOWN_POKEMON_CENTER_1F),
+        .map = MAP_NUM(VERDANTURF_TOWN_POKEMON_CENTER_1F),
+        DEFAULT_POKEMON_CENTER_COORDS,
+    },
+    [HEAL_LOCATION_PACIFIDLOG_TOWN - 1] =
+    {
+        .group = MAP_GROUP(PACIFIDLOG_TOWN_POKEMON_CENTER_1F),
+        .map = MAP_NUM(PACIFIDLOG_TOWN_POKEMON_CENTER_1F),
+        DEFAULT_POKEMON_CENTER_COORDS,
+    },
+    [HEAL_LOCATION_EVER_GRANDE_CITY_POKEMON_LEAGUE - 1] =
+    {
+        .group = MAP_GROUP(EVER_GRANDE_CITY_POKEMON_LEAGUE_1F),
+        .map = MAP_NUM(EVER_GRANDE_CITY_POKEMON_LEAGUE_1F),
+        .x = 3,
+        .y = 4,
+    },
+    [HEAL_LOCATION_SOUTHERN_ISLAND_EXTERIOR - 1] =
+    {
+        .group = MAP_GROUP(SOUTHERN_ISLAND_EXTERIOR),
+        .map = MAP_NUM(SOUTHERN_ISLAND_EXTERIOR),
+        .x = 15,
+        .y = 20,
+    },
+    [HEAL_LOCATION_BATTLE_FRONTIER_OUTSIDE_EAST - 1] =
+    {
+        .group = MAP_GROUP(BATTLE_FRONTIER_POKEMON_CENTER_1F),
+        .map = MAP_NUM(BATTLE_FRONTIER_POKEMON_CENTER_1F),
+        DEFAULT_POKEMON_CENTER_COORDS,
+    },
+};
+
+#undef DEFAULT_POKEMON_CENTER_COORDS
+
+static const u8 sWhiteoutRespawnHealerNpcLocalIds[HEAL_LOCATION_COUNT - 1] =
+{
+    [HEAL_LOCATION_LITTLEROOT_TOWN_BRENDANS_HOUSE_2F - 1] = 1,
```

Comment:
Same, but that's how FRLG handled it. Another way to handle it would be to get the `MapHeader` of the warp destination, iterate over the `objectEvents` and check if `graphicsId` is either `OBJ_EVENT_GFX_MOM` or `OBJ_EVENT_GFX_NURSE` to find the `localId`